gitlab/resource_gitlab_branch.go

```diff
@@ -0,0 +1,181 @@
+package gitlab
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+	"log"
+)
+
+func resourceGitlabBranch() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabBranchCreate,
+		Read:   resourceGitlabBranchRead,
+		Delete: resourceGitlabBranchDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"project": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"ref": {
```

Comment:
This should be required, with no default, in order to match the API. https://docs.gitlab.com/ee/api/branches.html#create-repository-branch Especially since newer repositories use `main` as a default.

Reply:
I have had to leave this as optional, it is not ideal but the reason being is there is no real way to determine the ref of the branch after creation as the api does not return this in either the branches or commit apis. This then makes it near impossible to import it while this is required with no default as it will always flag a new value and attempt to destroy the imported branch upon terraform apply. If there is a workaround please let me know but for now I have added "main" as the default value and reverted to it being an optional property.